st2api/st2api/controllers/v1/keyvalue.py

```diff
@@ -55,26 +59,37 @@ def __init__(self):
         self._coordinator = coordination.get_coordinator()
         self.get_one_db_method = self._get_by_name
 
-    @jsexpose(arg_types=[str, str, bool])
-    def get_one(self, name, scope=SYSTEM_SCOPE, decrypt=False):
+    @jsexpose(arg_types=[str, str, str, bool])
+    def get_one(self, name, scope=SYSTEM_SCOPE, user=None, decrypt=False):
         """
             List key by name.
 
             Handle:
                 GET /keys/key1
         """
         self._validate_scope(scope=scope)
-        key_ref = get_key_reference(scope=scope, name=name, user=get_requester())
+
+        if user:
+            # Providing a user implies a user scope
+            scope = USER_SCOPE
+
+        requester_user = get_requester()
+        user = user or requester_user
+        is_admin = request_user_is_admin(request=pecan.request)
+
+        # User needs to be either admin or requesting item for itself
+        self._validate_decrypt_query_parameter(decrypt=decrypt, scope=scope, is_admin=is_admin)
+
+        # Validate that the authenticated user is admin if user query param is provided
+        self._validate_user_query_parameter(user=user, is_admin=is_admin)
```

Comment:
You mean before `self._validate_decrypt_query_parameter` call?

If so, I don't think the order really matters that much.